### PR TITLE
xerces-c: Fix clang compilation issue on Windows

### LINKED ISSUE #24796: [package] xerces-c/3.2.5: Unable to compile with clang-cl and llvm-rc
### Description

Hello,

@uilianries 
I have been unable to compile xerces on windows using clang-cl. The issues actually comes from llvm-rc: https://github.com/llvm/llvm-project/issues/57367

The fix is pretty straight forward:

```cmake
if (WIN32 AND CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
    set(CMAKE_RC_FLAGS="-C 1252")
endif()
```

While I could inject the variable to my profile through `tools.cmake.cmaketoolchain:extra_variables` (it does fix the error), I think it would be better if added directly to the recipe. What do you reckon?

### Package and Environment Details

```powershell
version: 2.6.0
conan_path: D:\venv\Scripts\conan
python
  version: 3.9.13
  sys_version: 3.9.13 (tags/v3.9.13:6de2ca5, May 17 2022, 16:36:42) [MSC v.1929 64 bit (AMD64)]
  sys_executable: D:\venv\Scripts\python.exe
  is_frozen: False
  architecture: AMD64
system
  version: 10.0.22631
  platform: Windows-10-10.0.22631-SP0
  system: Windows
  release: 10
  cpu: Intel64 Family 6 Model 186 Stepping 3, GenuineIntel
```

### Conan profile

```txt
{% set compiler, version, compiler_exe = detect_api.detect_clang_compiler(compiler_exe="clang") %}
{% set version = detect_api.default_compiler_version(compiler, version) %}
{% set os = detect_api.detect_os() %}

[settings]
arch={{detect_api.detect_arch()}}
build_type=Release
compiler={{compiler}}
compiler.cppstd=17
compiler.version={{version}}
os={{os}}

[conf]
tools.cmake.cmaketoolchain:generator=Ninja
tools.cmake.cmake_layout:build_folder_vars = ["settings.os", "settings.compiler", "settings.compiler.version", "settings.arch"]
tools.build:compiler_executables = {"c": "clang-cl", "cpp": "clang-cl", "rc": "llvm-rc"}
tools.build:cxxflags=["-fansi-escape-codes -fcolor-diagnostics"]
tools.build:exelinkflags=["-fuse-ld=lld"]
tools.build:sharedlinkflags=["-fuse-ld=lld"]

{% if os == "Windows" %}
	mypkg*/*:tools.env.virtualenv:powershell=True
	tools.cmake.cmaketoolchain:generator=Ninja Multi-Config
{% endif %}
```

### Steps to reproduce

conanfile:
```txt
[requires]
xerces-c/3.2.5

[options]
xerces-c*:shared=True
xerces-c*:network=False
xerces-c*:char_type=char16_t
```
`conan install --profile:all clang --build=missing -verror --remote conancenter conanfile.txt`

### Logs

<details><summary>Click to expand log</summary>

```
[287/299] Building RC object src\CMakeFiles\xerces-c.dir\Release\xercesc\util\version.rc.res
FAILED: src/CMakeFiles/xerces-c.dir/Release/xercesc/util/version.rc.res 
"C:\Program Files\Microsoft Visual Studio\2022\Professional\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -E cmake_llvm_rc D:\conan\p\b\xerce57c5f34a11450\b\build\src\xercesc\util\version.rc src\CMakeFiles\xerces-c.dir\Release\xercesc\util\version.rc.res.pp C:\PROGRA~1\LLVM\bin\clang-cl.exe -DHAVE_CONFIG_H=1 -DXERCES_BUILDING_LIBRARY=1 -DXERCES_DLL_NAME=\"xerces-c_3_2.dll\0\" -D_THREAD_SAFE=1 -Dxerces_c_EXPORTS -DCMAKE_INTDIR=\"Release\" -DRC_INVOKED -I D:\conan\p\b\xerce57c5f34a11450\b\build -I D:\conan\p\b\xerce57c5f34a11450\b\src\src -I D:\conan\p\b\xerce57c5f34a11450\b\build\src -DWIN32 -clang:-MD -clang:-MF -clang:src\CMakeFiles\xerces-c.dir\Release\xercesc\util\version.rc.res.d  -E -- D:\conan\p\b\xerce57c5f34a11450\b\build\src\xercesc\util\version.rc ++ llvm-rc -DHAVE_CONFIG_H=1 -DXERCES_BUILDING_LIBRARY=1 -DXERCES_DLL_NAME=\"xerces-c_3_2.dll\0\" -D_THREAD_SAFE=1 -Dxerces_c_EXPORTS -DCMAKE_INTDIR=\"Release\" -I SOURCE_DIR -I D:\conan\p\b\xerce57c5f34a11450\b\build -I D:\conan\p\b\xerce57c5f34a11450\b\src\src -I D:\conan\p\b\xerce57c5f34a11450\b\build\src -DWIN32 -clang:-MD -clang:-MF -clang:src\CMakeFiles\xerces-c.dir\Release\xercesc\util\version.rc.res.d /fo src\CMakeFiles\xerces-c.dir\Release\xercesc\util\version.rc.res src\CMakeFiles\xerces-c.dir\Release\xercesc\util\version.rc.res.pp
llvm-rc: Error in VERSIONINFO statement (ID 1): 
Non-ASCII 8-bit codepoint (´┐¢) can't be interpreted in the current codepage

```

</details>

### PR DESCRIPTION
**Pull request by perseoGI**
### Summary
Changes to recipe:  **xerces-c**

#### Motivation

As explained in #24796, when compiling this package with `clang-cl` in Windows, it will through an error which is easily fixable defining `CMAKE_RC_FLAGS` variable.

#### Details

Close #24796

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Review by AbrilRBS — DISMISSED** [review]
Great job! :)

File: recipes/xerces-c/all/conanfile.py [hunk 1]
```diff
@@ -131,6 +131,12 @@ def generate(self):
         # avoid picking up system dependency
         tc.variables["CMAKE_DISABLE_FIND_PACKAGE_CURL"] = self.options.get_safe("network_accessor") != "curl"
         tc.variables["CMAKE_DISABLE_FIND_PACKAGE_ICU"] = "icu" not in (self.options.transcoder, self.options.message_loader)
+
+        # Fix compatibility with Clang on Windows
+        # https://issues.apache.org/jira/browse/XERCESC-2252
+        if self.settings.os == "Windows" and self.settings.compiler == "Clang":
```
Comment (AbrilRBS) [inline recipes/xerces-c/all/conanfile.py:137]:
```suggestion
        if self.settings.os == "Windows" and self.settings.compiler == "clang":
```

**Comment by valgur** [comment]
The

> Library './lib/libxerces-c.so' links to system library 'nsl' but it is not in cpp_info.system_libs.

hook warning is worth fixing as well, if you don't mind.

**Review by valgur — APPROVED** [review]

**Review by AbrilRBS — APPROVED** [review]

**Review by valgur — APPROVED** [review]

**Comment by jcar87** [comment]
https://github.com/conan-io/conan-center-index/issues/25502

Once again, it's worth mentioning that if the goal is to fix a clang issue on Windows - that the PR does not try to do other things. Given the size of the backlog, we want to minimise the risk of introducing any regressions.

> Library './lib/libxerces-c.so' links to system library 'nsl' but it is not in cpp_info.system_libs.

We need to work better on the wording on the hook warning, as it does *not* always mean that the solution is to simply add the mentioned library as a system dependency, and it's more a "please look into this". 
The check the hooks performs is there to warn us that "there's a chance you won't be able to link to the static version of this library, and if that is the case, this is probably why". The warning comes from the _shared_ variant of the package, but the potential error would happen when consuming the static one. If there are no errors (even for consumers), then the warning can potentially be disregarded.

Secondary to this, _if_ a library must be added to the system requirements, we need to verify _where_ it comes from, as it may not be part of a "vanilla" Linux system. Some are okay (pthread, dl, etc), as they are part of a typical runtime, but some may not be. If the library reported in the hook is not found by default on most linux distros, we need to check why and where it is needed, and if it _is_ needed, ensure that there are calls to the relevant system package managers if we don't (or can't) have a conan dependency for the particular library.

We'll do better on this.

**Pull request merged**